core/src/kura.rs

```diff
@@ -463,37 +465,48 @@ impl<L: Lock> BlockStore<L> {
     ///
     /// # Errors
     /// IO Error.
-    #[allow(clippy::needless_range_loop)]
+    #[warn(clippy::needless_range_loop)] // This could be a problem, because we might be truncating the blocks the wrong way around.
```

Comment:
What do you mean by this? 

Actually fix for this lint is rather simple, we replace:

```rust 
for index in 0..block_count {
    dest_buffer[index] = (
         // ...
    );
}
```

With: 

```rust
for current_buffer in dest_buffer.iter_mut() {
    *current_buffer = (
         // ...
    );
}
```

Reply:
I mean the logic of the operation is dangerous. And I'm afraid that block wouldn't do because of the coupling. There is a way to fix that lint, however, I wanted to focus on getting the code to work. Given that this wasn't merged, I'll draft and updraft once lints are fixed.